qemu/tests/virtio_console.py

```diff
@@ -563,12 +564,145 @@ def test_loopback():
+                        if isinstance(qdev, qemu_qtree.QtreeDev) and (
+                                      'id' in qdev.qtree) and (
+                                      qdev.qtree['id'] == qtree_check_port):
+                            return qbus.qtree['id']
+            return None
+        except AttributeError as err:
+            logging.warn("Monitor deson't supoort qtree skip this test, error"
+                         " log: %s " % err)
+
     @error_context.context_aware
     def test_interrupted_transfer():
         """
         This test creates loopback between 2 ports and interrupts transfer
         eg. by stopping the machine or by unplugging of the port.
         """
+        def _replug_loop():
+            """ Replug ports and pci in a loop """
+            def _port_unplug(port_idx):
+                dev = ports[port_idx]
+                portdev = get_virtio_port_by_name(vm, dev.name)
+                portdev = vm.devices.get_by_params({"name": dev.qemu_id})[0]
+                if not portdev:
+                    test.error("No port named %s" % dev.qemu_id)
+                if portdev.get_param("bus") == "None":
+                    bus_info = portdev.get_param("bus")
+                else:
+                    bus_info = get_bus_of_port(vm, portdev)
+                port_property = dict(id=portdev.get_param("id"),
+                                     name=portdev.get_param("name"),
+                                     chardev=portdev.get_param("chardev"),
+                                     bus=bus_info)
+                if not port_property["bus"]:
+                    test.error("Can't find the bus %s for this port" % bus_info)
+                (out, ver_out) = vm.devices.simple_unplug(portdev, vm.monitor)
+                if not ver_out:
+                    test.error("Error occured when unplug %s" % dev.name)
+                time.sleep(intr_time)
+                return port_property
+
+            def _port_plug(device, property):
+                portdev = qdevices.QDevice(device)
+                for key, value in {'id': property['id'],
+                                   'chardev': property['chardev'],
+                                   'name': property['name'],
+                                   'bus': property['bus']}.items():
+                    portdev.set_param(key, value)
+                (out, ver_out) = vm.devices.simple_hotplug(portdev, vm.monitor)
+                if not ver_out:
+                    test.error("Error occured when plug port %s." % property['name'])
+                time.sleep(intr_time)
+
+            def _pci_unplug(bus):
+                device = vm.devices.get_by_params({"id": str(bus).split('.')[0]})[0]
+                if not device:
+                    test.error("No bus %s in vm." % bus)
+                bus_property = dict(id=device.get_param("id"))
+                err = ""
+                out = device.unplug(vm.monitor)
```

Comment:
Now it should be okay to use `vm.devices.simple_unplug()`, there it is updated to try 30s for `verify_unplug()`